lib/ring_buffer/ring_buffer.c

```diff
@@ -79,16 +79,50 @@ int sys_ring_buf_get(struct ring_buf *buf, u16_t *type, u8_t *value,
 	if (likely(buf->mask)) {
 		for (i = 0; i < header->length; ++i) {
 			index = (i + buf->head + 1) & buf->mask;
-			data[i] = buf->buf[index];
+			data[i] = buf->buf.buf32[index];
 		}
 		buf->head = (buf->head + header->length + 1) & buf->mask;
 	} else {
 		for (i = 0; i < header->length; ++i) {
 			index = (i + buf->head + 1) % buf->size;
-			data[i] = buf->buf[index];
+			data[i] = buf->buf.buf32[index];
 		}
 		buf->head = (buf->head + header->length + 1) % buf->size;
 	}
 
 	return 0;
 }
+
+size_t sys_ring_buf_raw_put(struct ring_buf *buf, u8_t *data, size_t size)
+{
+	u32_t i, space, index;
+	size_t cpy_size;
+
+	space = sys_ring_buf_space_get(buf);
+	cpy_size = size > space ? space : size;
+
+	for (i = 0; i < cpy_size; i++) {
+		index = (i + buf->tail) % buf->size;
```

Comment:
Ok, so please don't use "expensive modulo operations", please use `if` to check for overflow instead.